### PR TITLE
pulseeffects: 4.8.4 -> 5.0.0, pulseeffects-legacy: init at 4.8.4

### PR DESCRIPTION
**Pull request by ymeister**
###### Motivation for this change

Version 5.0.0 released:
https://github.com/wwmm/pulseeffects/releases/tag/v5.0.0

As of this version PulseEffects uses PipeWire instead of PulseAudio. Because of the currently lesser adoption of PipeWire, legacy version of PulseEffects (pulseeffects-legacy) was added. Further updates of the legacy version can be acquired from corresponding branch https://github.com/wwmm/pulseeffects/tree/pulseaudio-legacy. Please note, that legacy branch is in maintenance mode and no longer under development.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/audio/pulseeffects/default.nix [hunk 1]
```diff
@@ -45,13 +45,13 @@ let
   ];
 in stdenv.mkDerivation rec {
   pname = "pulseeffects";
-  version = "4.8.4";
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/pulseeffects/default.nix:48]: Please put this into a separate commit.

File: pkgs/applications/audio/pulseeffects-legacy/default.nix [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+  # Meson is no longer able to pick up Boost automatically.
+  # https://github.com/NixOS/nixpkgs/issues/86131
+  BOOST_INCLUDEDIR = "${lib.getDev boost}/include";
+  BOOST_LIBRARYDIR = "${lib.getLib boost}/lib";
+
+  meta = with lib; {
+    description = "Limiter, compressor, reverberation, equalizer and auto volume effects for Pulseaudio applications";
+    homepage = "https://github.com/wwmm/pulseeffects";
+    license = licenses.gpl3;
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/pulseeffects-legacy/default.nix:110]: gpl3Plus or gpl3Only?

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 112591` at f979af82 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>pulseeffects</li>
    <li>pulseeffects-legacy</li>
  </ul>
</details>

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 112591` at f979af82 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li><a href="https://gist.github.com/eb49b59eefaf4b2eed0a8f66f50b24c0">pulseeffects</a></li></li>
    <li><a href="https://gist.github.com/550a42092fde3667c6a22ae491f9623a">pulseeffects-legacy</a></li></li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl3` is a deprecated license, check if project uses `gpl3Plus` or `gpl3Only` and change `meta.license` accordingly.

Near pkgs/applications/audio/pulseeffects-legacy/default.nix:110:5:
```
    |
110 |     license = licenses.gpl3;
    |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl3` is a deprecated license, check if project uses `gpl3Plus` or `gpl3Only` and change `meta.license` accordingly.

Near pkgs/applications/audio/pulseeffects/default.nix:110:5:
```
    |
110 |     license = licenses.gpl3;
    |     ^
```
</li>
  </ul>
</details>

File: pkgs/applications/audio/pulseeffects-legacy/default.nix (hunk 2 above)
Comment (ymeister) [reply pkgs/applications/audio/pulseeffects-legacy/default.nix:110]: I'm not familiar with this topic. Could you please check yourself? (Don't want to accidentally make any mistakes)
https://github.com/wwmm/pulseeffects/blob/master/LICENSE.md

Comment (rmcgibbo) [reply pkgs/applications/audio/pulseeffects-legacy/default.nix:110]: https://github.com/wwmm/pulseeffects/blob/1bb9be071bcf0c1f89b67a7e4eb8d3405dcb8a18/include/autogain.hpp#L9 looks like gpl3Plus

Comment (ymeister) [reply pkgs/applications/audio/pulseeffects-legacy/default.nix]: So be it.

**Comment by jtojnar** [comment]
Does this still need `gst-plugins-good` now that it switched from `pulsesrc` to `pipewiresrc`? Could you to remove it, test pulseeffects and check the output in the terminal for any messages about missing GStreamer elements?

**Comment by ymeister** [comment]
> Does this still need `gst-plugins-good` now that it switched from `pulsesrc` to `pipewiresrc`? Could you try to remove it, test pulseeffects and check the output in the terminal for any messages about missing GStreamer elements?

Still produces an error, although in slightly different form:
```
(pulseeffects:5480): glibmm-ERROR **: 06:28:23.639:
unhandled exception (type std::exception) in signal handler:
what: soe: Failed to get required plugin: spectrum

Trace/breakpoint trap (core dumped)
```

**Comment by jtojnar** [comment]
Looks like `spectrum` is [part of the good plug-ins](https://gstreamer.freedesktop.org/documentation/spectrum/index.html?gi-language=c#factory-details). Could you update the comment accordingly?

**Review by jtojnar — APPROVED** [review]
Changes look good to me but I cannot test it with pipewire-pulse at the moment.

**Comment by jtojnar** [comment]
After IRC discussion, I have decided to deprecate the old attribute name to avoid people being confused why it crashes: see https://github.com/wwmm/pulseeffects/issues/891 and https://github.com/wwmm/pulseeffects/issues/874
